File: Android/MeetingManagerAndroid/app/src/main/java/com/microsoft/office365/meetingmgr/Models/FileAttachment.java
/*
 * Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license.
 * See LICENSE in the project root for license information.
 */
package com.microsoft.office365.meetingmgr.Models;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Representation of file attachment
 */
public class FileAttachment {
    @JsonProperty("@odata.type")
    public String Type = "#Microsoft.OutlookServices.FileAttachment";
    public final String ContentBytes;
    public final String Name;

    private FileAttachment() {
        this(null, null);
    }

    public FileAttachment(String content, String name) {
        ContentBytes = content;
        Name = name;
    }
}

File: Xamarin/Meeting_Manager_Xamarin/Meeting_Manager_Xamarin/Meeting_Manager_Xamarin/Models/FileAttachment.cs
//Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license.
//See LICENSE in the project root for license information.

using System;

namespace Meeting_Manager_Xamarin.Models
{
    public class FileAttachment
    {
        [Newtonsoft.Json.JsonProperty("@odata.type")]
        public string Type => "#microsoft.graph.fileAttachment";

        public string Id { get; set; }
        public string Name { get; set; }
        public Int32? Size { get; set; }
        public string ContentType { get; set; }
        public string ContentLocation { get; set; }
        public byte[] ContentBytes { get; set; }
        public bool IsInline { get; set; }
    }
}
